extensions/adapters/raster/src/main/java/org/locationtech/geowave/adapter/raster/operations/DeletePyramidLevelCommand.java

```diff
@@ -0,0 +1,222 @@
+import org.locationtech.geowave.core.store.adapter.InternalDataAdapter;
+import org.locationtech.geowave.core.store.adapter.statistics.DataStatisticsStore;
+import org.locationtech.geowave.core.store.adapter.statistics.InternalDataStatistics;
+import org.locationtech.geowave.core.store.adapter.statistics.PartitionStatistics;
+import org.locationtech.geowave.core.store.api.DataStore;
+import org.locationtech.geowave.core.store.api.DataTypeAdapter;
+import org.locationtech.geowave.core.store.api.Index;
+import org.locationtech.geowave.core.store.api.QueryBuilder;
+import org.locationtech.geowave.core.store.cli.store.DataStorePluginOptions;
+import org.locationtech.geowave.core.store.cli.store.StoreLoader;
+import org.locationtech.geowave.core.store.util.CompoundHierarchicalIndexStrategyWrapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import com.beust.jcommander.Parameter;
+import com.beust.jcommander.ParameterException;
+import com.beust.jcommander.Parameters;
+
+@GeowaveOperation(name = "deletelevel", parentOperation = RasterSection.class)
+@Parameters(commandDescription = "Delete a pyramid level of a raster layer")
+public class DeletePyramidLevelCommand extends DefaultOperation implements Command {
+  private static final Logger LOGGER = LoggerFactory.getLogger(DeletePyramidLevelCommand.class);
+  @Parameter(description = "<store name>")
+  private List<String> parameters = new ArrayList<>();
+
+  @Parameter(names = "--level", description = "The raster pyramid level to delete", required = true)
+  private final Integer level = null;
+  @Parameter(
+      names = "--coverage",
+      description = "The raster coverage name (required if store has multiple coverages)")
+  private final String coverageName = null;
+
+  private DataStorePluginOptions inputStoreOptions = null;
+
+
+  @Override
+  public void execute(final OperationParams params) throws Exception {
+    run(params);
+  }
+
+  public void run(final OperationParams params) {
+    // Ensure we have all the required arguments
+    if (parameters.size() != 1) {
+      throw new ParameterException("Requires argument: <store name>");
+    }
+
+    final String inputStoreName = parameters.get(0);
+
+    // Config file
+    final File configFile = getGeoWaveConfigFile(params);
+
+    // Attempt to load input store.
+    final StoreLoader inputStoreLoader = new StoreLoader(inputStoreName);
+    if (!inputStoreLoader.loadFromConfig(configFile)) {
+      throw new ParameterException("Cannot find store name: " + inputStoreLoader.getStoreName());
+    }
+    inputStoreOptions = inputStoreLoader.getDataStorePlugin();
+    final DataStore store = inputStoreLoader.createDataStore();
+    RasterDataAdapter adapter = null;
+
+    for (final DataTypeAdapter<?> type : store.getTypes()) {
+      if (isRaster(type)
+          && ((coverageName == null) || coverageName.equals(adapter.getTypeName()))) {
+        if (adapter != null) {
+          LOGGER.error(
```

Comment:
There is some inconsistency with the way status/errors are being displayed to the user.  In other CLI commands errors are typically reported via ParameterException, with information type text being reported via `JCommander.getConsole().println`.  I think the error logging should be updated and perhaps the `LOGGER.info` parts as well, but as I said before, there seems to be some inconsistency amongst existing commands.

Reply:
I looked at this and I think there's a couple ways to look at it - always throw an exception on errors would be an option and let GeoWaveMain take care of handling exceptional cases, or always use JCommander.getConsole().println() although because a lot of logic in geowave commands is potentially shared with REST and gRPC services this is probably not the right approach.  It seems actually we should be moving more to some generalized logging that just happens to be to JCommander console from the CLI but is logging to file for the services.  Regardless, I think whatever is done, it may as well be done consistently outside of the scope of this.